packages/@aws-cdk/core/lib/tag-manager.ts

```diff
@@ -215,18 +239,27 @@ export class TagManager {
     return (construct as any).tags !== undefined;
   }
 
+  /**
+   * The property name for tag values
+   *
+   * Normally this is `tags` but some resources choose a different name. Cognito
+   * UserPool uses UserPoolTags
+   */
+  public readonly tagPropertyName: string;
+
   private readonly tags = new Map<string, Tag>();
   private readonly priorities = new Map<string, number>();
   private readonly tagFormatter: ITagFormatter;
   private readonly resourceTypeName: string;
   private readonly initialTagPriority = 50;
 
-  constructor(tagType: TagType, resourceTypeName: string, tagStructure?: any) {
```

Comment:
I don't think we can break this API. It's marked as stable.

You can add a fourth optional argument with `TagManagerOptions` and add the the 2.0 issue a note to clean this up later.

Reply:
yep -- change coming up.